govukurllookup.py

```diff
@@ -31,36 +37,45 @@ def lookup(self):
 
         return self.urldicts
 
-def api_lookup(x):
+def api_lookup(url):
     
     '''
-    Simple function to lookup a url on the GOV.UK content API
-    Takes as an input the dictionary output by clean_url()
+    Lookup a url on the GOV.UK content API
+
+    Take a single url string as input, and returns a dict returned by an api
+    call to the content api.
     '''
+    # Form the api url
+
+    api_url = "https://www.gov.uk/api/content{}".format(url)
 
-    url = "https://www.gov.uk/api/content" + x
-    
     try:
-       
-        # read JSON result into r
-        r = requests.get(url)
-        results = r.json()
+
+        # Lookup the url and return the json
+
+        results = requests.get(api_url).json()
+
+        # Check whether api returned a redirect, and if so look up the api_url
+        # using a standard http call so that we are returned the redirect url.
 
         if results['document_type'] == "redirect":
-            url = "https://www.gov.uk" + x
-            s = requests.get(url)
-            redirected_url = s.url
-            redirected_api =redirected_url.replace("https://www.gov.uk", 
-                "https://www.gov.uk/api/content")
-            r = requests.get(redirected_api)
-            results = r.json()
+            redirect_url = "https://www.gov.uk" + url
```

Comment:
to be honest, there are so many ways to concatenate text in python. I read a blog somewhere that said that the `{}.format()` method is the one to use, but I couldn't tell you why!